packages/generator/src/util/test/compileThis.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+
+export function compileThis(files: Record<string, string>, basePath: string) {
+  const compilerOptions: ts.CompilerOptions = {
+    noEmit: true,
+    target: ts.ScriptTarget.ES2020,
```

Comment:
we might want to make the generated typescript super strict so that anyone with lib type checking doesn't complain that we broke their build

Reply:
Hmm. Do you have an example how this would manifest?

Reply:
Reaching back into the cobweb-covered part of my brain I think its related to maybe when we turned on strict null checks? I think at some point there was a way for a library to be compiled with some insufficient level of strictness that could sometimes trip up consumers who wanted to enable the new strict modes before their dependencies got around to it. Maybe this just isn't a real concern in 2023.

https://github.com/microsoft/TypeScript/issues/14810#issuecomment-288811529